File: java_projects/glue_code/commons-graph/src/main/java/org/apache/commons/graph/shortestpath/HeuristicBuilder.java
package org.apache.commons.graph.shortestpath;

import org.apache.commons.graph.WeightedPath;

public interface HeuristicBuilder<V, WE, W> {
  <H extends Heuristic<V, W>> WeightedPath<V, WE, W> withHeuristic(H heuristic);
}

File: data/skeletons/commons-graph/src/main/org/apache/commons/graph/shortestpath/HeuristicBuilder.py
# Imports Begin
from src.main.org.apache.commons.graph.shortestpath.Heuristic import *
from src.main.org.apache.commons.graph.WeightedPath import *
import typing
import pathlib
from abc import ABC

# Imports End


class HeuristicBuilder(ABC):

    # Class Fields Begin
    # Class Fields End

    # Class Methods Begin
    def withHeuristic(
        self, heuristic: typing.Any
    ) -> WeightedPath[typing.Any, typing.Any, typing.Any]:
        pass

    # Class Methods End
